### ISSUE TITLE
Doesn't work with Sonarqube version 8.7.0

### ISSUE THREAD

**Issue by alekberovvusal** [mc1arke/sonarqube-community-branch-plugin#309]
Here is error message. 
Plugin version: 1.6.0
```
2021.02.25 13:55:55 ERROR web[][o.s.s.p.Platform] Background initialization failed. Stopping SonarQube
java.lang.RuntimeException: Failed to load core extension Community Branch Plugin
	at org.sonar.core.extension.CoreExtensionsInstaller.install(CoreExtensionsInstaller.java:83)
	at org.sonar.core.extension.CoreExtensionsInstaller.lambda$install$2(CoreExtensionsInstaller.java:71)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.sonar.core.extension.CoreExtensionsInstaller.install(CoreExtensionsInstaller.java:71)
	at org.sonar.server.platform.platformlevel.PlatformLevel4.start(PlatformLevel4.java:572)
	at org.sonar.server.platform.PlatformImpl.start(PlatformImpl.java:213)
	at org.sonar.server.platform.PlatformImpl.startLevel34Containers(PlatformImpl.java:187)
	at org.sonar.server.platform.PlatformImpl.access$500(PlatformImpl.java:46)
	at org.sonar.server.platform.PlatformImpl$1.lambda$doRun$0(PlatformImpl.java:120)
	at org.sonar.server.platform.PlatformImpl$AutoStarterRunnable.runIfNotAborted(PlatformImpl.java:370)
	at org.sonar.server.platform.PlatformImpl$1.doRun(PlatformImpl.java:120)
	at org.sonar.server.platform.PlatformImpl$AutoStarterRunnable.run(PlatformImpl.java:354)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: Unable to register extension com.github.mc1arke.sonarqube.plugin.server.CommunityBranchSupportDelegate
	at org.sonar.core.platform.ComponentContainer.addExtension(ComponentContainer.java:254)
	at org.sonar.core.extension.CoreExtensionsInstaller.addSupportedExtension(CoreExtensionsInstaller.java:118)
	at org.sonar.core.extension.CoreExtensionsInstaller.access$100(CoreExtensionsInstaller.java:40)
	at org.sonar.core.extension.CoreExtensionsInstaller$ContextImpl.addExtension(CoreExtensionsInstaller.java:161)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at org.sonar.core.extension.CoreExtensionsInstaller$ContextImpl.addExtensions(CoreExtensionsInstaller.java:172)
	at com.github.mc1arke.sonarqube.plugin.CommunityBranchPlugin.load(CommunityBranchPlugin.java:75)
	at org.sonar.core.extension.CoreExtensionsInstaller.addDeclaredExtensions(CoreExtensionsInstaller.java:90)
	at org.sonar.core.extension.CoreExtensionsInstaller.install(CoreExtensionsInstaller.java:77)
	... 14 common frames omitted
Caused by: java.lang.NoClassDefFoundError: org/sonar/db/organization/OrganizationDto
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
	at org.picocontainer.injectors.AdaptingInjection$1.run(AdaptingInjection.java:203)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at org.picocontainer.injectors.AdaptingInjection.injectionMethodAnnotated(AdaptingInjection.java:200)
	at org.picocontainer.injectors.AdaptingInjection.methodAnnotatedInjectionAdapter(AdaptingInjection.java:171)
	at org.picocontainer.injectors.AdaptingInjection.createComponentAdapter(AdaptingInjection.java:70)
	at org.picocontainer.behaviors.AbstractBehaviorFactory.createComponentAdapter(AbstractBehaviorFactory.java:44)
	at org.picocontainer.behaviors.OptInCaching.createComponentAdapter(OptInCaching.java:45)
	at org.picocontainer.DefaultPicoContainer.addComponent(DefaultPicoContainer.java:536)
	at org.picocontainer.DefaultPicoContainer.access$300(DefaultPicoContainer.java:84)
	at org.picocontainer.DefaultPicoContainer$AsPropertiesPicoContainer.addComponent(DefaultPicoContainer.java:1149)
	at org.sonar.core.platform.ComponentContainer.addExtension(ComponentContainer.java:252)
	... 23 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.sonar.db.organization.OrganizationDto
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 37 common frames omitted
2021.02.25 13:55:55 INFO  web[][o.s.p.ProcessEntryPoint] Hard stopping process
```

**Comment by bmaehr** [comment]
Master did work for me with sonar 8.7 and gitlab after fixing the configuration. It is really annoying that SonarSource is doing so much refactoring.

I would suggest to change https://github.com/mc1arke/sonarqube-community-branch-plugin/blob/9970d5c704c5f4ece920b7e909d71c90f79377bc/src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/PullRequestPostAnalysisTask.java#L103 to info to see the message in log file when searching for the problem.

**Comment by blueghost1983** [comment]
It seems there're still some problem with pull request, got this error when handling pull request (merging to the main branch)
Error Details
java.lang.IllegalStateException: Could not find target branch 'null' in project
	at com.github.mc1arke.sonarqube.plugin.ce.CommunityBranchLoaderDelegate.createPullRequest(CommunityBranchLoaderDelegate.java:98)
	at com.github.mc1arke.sonarqube.plugin.ce.CommunityBranchLoaderDelegate.load(CommunityBranchLoaderDelegate.java:78)
	at com.github.mc1arke.sonarqube.plugin.ce.CommunityBranchLoaderDelegate.load(CommunityBranchLoaderDelegate.java:50)
	at org.sonar.ce.task.projectanalysis.component.BranchLoader.load(BranchLoader.java:44)
	at org.sonar.ce.task.projectanalysis.step.LoadReportAnalysisMetadataHolderStep.execute(LoadReportAnalysisMetadataHolderStep.java:74)
	at org.sonar.ce.task.step.ComputationStepExecutor.executeStep(ComputationStepExecutor.java:81)
	at org.sonar.ce.task.step.ComputationStepExecutor.executeSteps(ComputationStepExecutor.java:72)
	at org.sonar.ce.task.step.ComputationStepExecutor.execute(ComputationStepExecutor.java:59)
	at org.sonar.ce.task.projectanalysis.taskprocessor.ReportTaskProcessor.process(ReportTaskProcessor.java:81)
	at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.executeTask(CeWorkerImpl.java:235)
	at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.run(CeWorkerImpl.java:217)
	at org.sonar.ce.taskprocessor.CeWorkerImpl.findAndProcessTask(CeWorkerImpl.java:162)
	at org.sonar.ce.taskprocessor.CeWorkerImpl$TrackRunningState.get(CeWorkerImpl.java:137)
	at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:89)
	at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:53)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

**Comment by mc1arke** [comment]
@blueghost1983 this doesn't contain enough information for this to be diagnosed. I suspect you've attempted to target a branch that Sonarqube doesn't actually know about, but without some information as to what parameters you're setting in the scanner, and what branches are currently visible in Sonarqube then I can't do any investigation.

**Comment by blueghost1983** [comment]
@mc1arke sorry for not containing the log. That was the error message I saw in the background tasks.  The code is in Azure DevOps pipeline, it happens when merging the branch to the develop branch, a pull request triggers the CI pipeline, and the error is when publish quality gate result. I now find the error in the logs/ce.log.
```
2021.03.11 09:41:43 INFO  ce[][o.s.c.t.CeWorkerImpl] Execute task | project=PROJECT | type=REPORT | id=AXggqdN0viU45gkESHOr | submitter=user1234
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Extract report | status=SUCCESS | time=205ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist scanner context | status=SUCCESS | time=7ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Propagate analysis warnings from scanner report | status=SUCCESS | time=0ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Generate analysis UUID | status=SUCCESS | time=0ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Load analysis metadata | status=SUCCESS | time=4ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Initialize | status=SUCCESS | time=0ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Build tree of components | components=327 | status=SUCCESS | time=19ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Validate project | status=SUCCESS | time=2ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Load quality profiles | status=SUCCESS | time=286ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Load Quality gate | status=SUCCESS | time=5ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Load new code period | status=SUCCESS | time=2ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Detect file moves | status=SUCCESS | time=0ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Load duplications | duplications=33 | status=SUCCESS | time=9ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute cross project duplications | status=SUCCESS | time=0ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute size measures | status=SUCCESS | time=15ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute new coverage | status=SUCCESS | time=1ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute coverage measures | status=SUCCESS | time=53ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute comment measures | status=SUCCESS | time=7ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Copy custom measures | status=SUCCESS | time=2ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute duplication measures | status=SUCCESS | time=1ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute size measures on new code | status=SUCCESS | time=0ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute language distribution | status=SUCCESS | time=3ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute test measures | status=SUCCESS | time=11ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute complexity measures | status=SUCCESS | time=2ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Load measure computers | status=SUCCESS | time=0ms
2021.03.11 09:41:44 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute Quality Profile status | status=SUCCESS | time=2ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Execute component visitors | status=SUCCESS | time=831ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Checks executed after computation of measures | status=SUCCESS | time=0ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute Quality Gate measures | status=SUCCESS | time=0ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Compute Quality profile measures | status=SUCCESS | time=1ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Generate Quality profile events | status=SUCCESS | time=1ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Generate Quality gate events | status=SUCCESS | time=1ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Check upgrade possibility for not analyzed code files. | status=SUCCESS | time=0ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist components | status=SUCCESS | time=140ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist analysis | status=SUCCESS | time=4ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist analysis properties | status=SUCCESS | time=0ms
2021.03.11 09:41:45 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist measures | inserts=50 | status=SUCCESS | time=7ms
2021.03.11 09:41:47 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist live measures | insertsOrUpdates=9574 | status=SUCCESS | time=2105ms
2021.03.11 09:41:47 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist duplication data | insertsOrUpdates=0 | status=SUCCESS | time=8ms
2021.03.11 09:41:47 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist new ad hoc Rules | status=SUCCESS | time=149ms
2021.03.11 09:41:48 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist issues | cacheSize=167 KB | inserts=274 | updates=0 | merged=0 | status=SUCCESS | time=78ms
2021.03.11 09:41:48 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist project links | status=SUCCESS | time=1ms
2021.03.11 09:41:48 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist events | status=SUCCESS | time=5ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist sources | status=SUCCESS | time=1047ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Persist cross project duplications | status=SUCCESS | time=0ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Enable analysis | status=SUCCESS | time=4ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Update last usage date of quality profiles | status=SUCCESS | time=9ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Purge db | status=SUCCESS | time=18ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Index analysis | status=SUCCESS | time=396ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Update need issue sync for branch | status=SUCCESS | time=3ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Send issue notifications | newIssuesNotifs=0 | newIssuesDeliveries=0 | myNewIssuesNotifs=0 | myNewIssuesDeliveries=0 | changesNotifs=0 | changesDeliveries=0 | status=SUCCESS | time=34ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Publish task results | status=SUCCESS | time=0ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.s.ComputationStepExecutor] Trigger refresh of Portfolios and Applications | status=SUCCESS | time=0ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.p.a.p.PostProjectAnalysisTasksExecutor] Webhooks | globalWebhooks=0 | projectWebhooks=0 | status=SUCCESS | time=7ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.p.a.p.PostProjectAnalysisTasksExecutor] Pull Request Decoration | status=SUCCESS | time=0ms
2021.03.11 09:41:49 INFO  ce[AXggqdN0viU45gkESHOr][o.s.c.t.CeWorkerImpl] Executed task | project=OVD-Admin | type=REPORT | id=AXggqdN0viU45gkESHOr | submitter=li-eric-liu40688 | status=SUCCESS | time=5822ms
2021.03.11 09:42:15 INFO  ce[][o.s.c.t.CeWorkerImpl] Execute task | project=MyAtlas_VCS | type=REPORT | pullRequest=91236 | id=AXggqktQviU45gkESHOs | submitter=nan-xin-zoey-xue96470
2021.03.11 09:42:16 INFO  ce[AXggqktQviU45gkESHOs][o.s.c.t.s.ComputationStepExecutor] Extract report | status=SUCCESS | time=58ms
2021.03.11 09:42:16 INFO  ce[AXggqktQviU45gkESHOs][o.s.c.t.s.ComputationStepExecutor] Persist scanner context | status=SUCCESS | time=5ms
2021.03.11 09:42:16 INFO  ce[AXggqktQviU45gkESHOs][o.s.c.t.s.ComputationStepExecutor] Propagate analysis warnings from scanner report | status=SUCCESS | time=0ms
2021.03.11 09:42:16 INFO  ce[AXggqktQviU45gkESHOs][o.s.c.t.s.ComputationStepExecutor] Generate analysis UUID | status=SUCCESS | time=0ms
2021.03.11 09:42:16 INFO  ce[AXggqktQviU45gkESHOs][o.s.c.t.s.ComputationStepExecutor] Load analysis metadata | status=FAILED | time=3ms
2021.03.11 09:42:16 ERROR ce[AXggqktQviU45gkESHOs][o.s.c.t.s.ComputationStepExecutor] Execution of listener failed
java.lang.IllegalStateException: Branch has not been set
        at com.google.common.base.Preconditions.checkState(Preconditions.java:508)
        at org.sonar.ce.task.projectanalysis.analysis.AnalysisMetadataHolderImpl.getBranch(AnalysisMetadataHolderImpl.java:150)
        at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.createBranch(PostProjectAnalysisTasksExecutor.java:222)
        at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.createProjectAnalysis(PostProjectAnalysisTasksExecutor.java:175)
        at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.finished(PostProjectAnalysisTasksExecutor.java:107)
        at org.sonar.ce.task.step.ComputationStepExecutor.executeListener(ComputationStepExecutor.java:91)
        at org.sonar.ce.task.step.ComputationStepExecutor.execute(ComputationStepExecutor.java:63)
        at org.sonar.ce.task.projectanalysis.taskprocessor.ReportTaskProcessor.process(ReportTaskProcessor.java:81)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.executeTask(CeWorkerImpl.java:235)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.run(CeWorkerImpl.java:217)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.findAndProcessTask(CeWorkerImpl.java:162)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$TrackRunningState.get(CeWorkerImpl.java:137)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:89)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:53)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
2021.03.11 09:42:16 ERROR ce[AXggqktQviU45gkESHOs][o.s.c.t.CeWorkerImpl] Failed to execute task AXggqktQviU45gkESHOs
java.lang.IllegalStateException: Could not find target branch 'null' in project
        at com.github.mc1arke.sonarqube.plugin.ce.CommunityBranchLoaderDelegate.createPullRequest(CommunityBranchLoaderDelegate.java:98)
        at com.github.mc1arke.sonarqube.plugin.ce.CommunityBranchLoaderDelegate.load(CommunityBranchLoaderDelegate.java:78)
        at com.github.mc1arke.sonarqube.plugin.ce.CommunityBranchLoaderDelegate.load(CommunityBranchLoaderDelegate.java:50)
        at org.sonar.ce.task.projectanalysis.component.BranchLoader.load(BranchLoader.java:44)
        at org.sonar.ce.task.projectanalysis.step.LoadReportAnalysisMetadataHolderStep.execute(LoadReportAnalysisMetadataHolderStep.java:74)
        at org.sonar.ce.task.step.ComputationStepExecutor.executeStep(ComputationStepExecutor.java:81)
        at org.sonar.ce.task.step.ComputationStepExecutor.executeSteps(ComputationStepExecutor.java:72)
        at org.sonar.ce.task.step.ComputationStepExecutor.execute(ComputationStepExecutor.java:59)
        at org.sonar.ce.task.projectanalysis.taskprocessor.ReportTaskProcessor.process(ReportTaskProcessor.java:81)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.executeTask(CeWorkerImpl.java:235)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.run(CeWorkerImpl.java:217)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.findAndProcessTask(CeWorkerImpl.java:162)
        at org.sonar.ce.taskprocessor.CeWorkerImpl$TrackRunningState.get(CeWorkerImpl.java:137)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:89)
        at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:53)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
2021.03.11 09:42:16 INFO  ce[AXggqktQviU45gkESHOs][o.s.c.t.CeWorkerImpl] Executed task | project=MyAtlas_VCS | type=REPORT | pullRequest=91236 | id=AXggqktQviU45gkESHOs | submitter=nan-xin-zoey-xue96470 | status=FAILED | time=181ms
```

**Comment by blueghost1983** [comment]
@mc1arke I finally found why I got the error. The default branch in azure devops repo is 'develop', but in sonarqube it is 'master', when pull request merges the branch to the 'develop' branch, the target branch can't be found in sonarqube. The solution is to change the 'master' branch name to same as default branch name in azure devops repo. anyway, thanks!

**Comment by bmaehr** [comment]
I was able to add settings on the master branch, e.g. 
```
`context.addExtensions(PropertyDefinition.builder(GitlabServerPullRequestDecorator.PULLREQUEST_CAN_FAIL_PIPELINE_ENABLED)`
                                          .category(CoreProperties.CATEGORY_GENERAL)
                                          .subCategory(CoreProperties.SUBCATEGORY_GENERAL)
                                          .onQualifiers(Qualifiers.PROJECT)
                                          .name("Fail pipeline if gate not reached")
                                          .description("Fail the pipeline if the Qualitiy Gate not passed succesfully.")
                                          .type(PropertyType.BOOLEAN)
                                          .defaultValue("true")
                                          .build());
```
it is displayed in the sonar UI, but when I do this `analysis.getScannerProperty(PULLREQUEST_CAN_FAIL_PIPELINE_ENABLED)` returns null.

**Comment by stevehipwell** [comment]
I'm having the same error on v8.7.0.

**Comment by Toilal** [comment]
It works when building from master branch.

**Comment by stevehipwell** [comment]
It does indeed work from master and I forked the repo to create a build for automation.

https://github.com/stevehipwell/sonarqube-community-branch-plugin/releases/tag/1.7.0-SNAPSHOT

**Comment by tcolgate** [comment]
I think the bug mentioned is a pre-existing bug, at least one reported for some non-github VCSs. I do get the error if a PR is targetting a branch other than master (current default branch), but PRs to master work fine.

**Comment by yfrindel** [comment]
Hi,

Do you know if sonarqube-community-branch-plugin work at least with Sonarqube version 8.6.0 please?

Regards

**Comment by rastislav-panis** [comment]
it is working on 8.6.1-community, so should be still working on 8.6.0.

**Comment by chly1103** [comment]
**Version：Sonarqube 8.7.1+sonarqube-community-branch-plugin-1.7.0-SNAPSHOT.jar**

mvn sonar:sonar -Dsonar.projectKey=*** -Dsonar.projectName=*** -Dsonar.projectVersion=master -Dsonar.ws.timeout=30 -Dsonar.login=[MASKED] -Dsonar.sourceEncoding=UTF-8 -Dsonar.java.binaries=target/classes -Dsonar.java.test.binaries=target/test-classes -Dsonar.java.surefire.report=target/surefire-reports -Dsonar.test.inclusions=/test/ -Dsonar.exclusions=/test/ --settings=./settings.xml -Dsonar.branch.name=master

[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.8.0.2131:sonar (default-cli) on project ils-parent: Failed to upload report - An error has occurred. Please contact your administrator -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

### **Successful execution without parameter -Dsonar.branch.name=master**

**Comment by mc1arke** [comment]
Fixed in 1.7.0 of the plugin

**Comment by albertinisg** [comment]
I got crazy with this error due to the Azure DevOps new naming from `master` to `main`. This drove me crazy and comment from @blueghost1983 opened my eyes. Just changed the branch name in `https://mysonarqube/project/branches?id=<projectName>` and it worked straight away!
